lib/pups/config.rb

```diff
@@ -3,24 +3,28 @@
 module Pups
   class Config
     attr_reader :config, :params
+    attr_accessor :ignored
 
-    def self.load_file(config_file)
-      new YAML.load_file(config_file)
+    def self.load_file(config_file, ignored = nil)
+      Config.new(YAML.load_file(config_file), ignored)
     rescue Exception
       warn "Failed to parse #{config_file}"
       warn "This is probably a formatting error in #{config_file}"
       warn "Cannot continue. Edit #{config_file} and try again."
       raise
     end
 
-    def self.load_config(config)
-      new YAML.safe_load(config)
+    def self.load_config(config, ignored = nil)
+      Config.new(YAML.safe_load(config), ignored)
     end
 
-    def initialize(config)
+    def initialize(config, ignored = nil)
       @config = config
       validate!(@config)
 
+      # remove any ignored config elements prior to any more processing
+      ignored&.each { |e| @config.delete(e) }
```

Comment:
Does `OptionParser` automatically split `--ignore env,params` into an array of `['env', 'params'`]?

Reply:
Also you could just default to an empty array instead of `nil`, then you don't need safe navigation

Reply:
Yes OptionParser automatically turns it into an array (this is covered in the specs at `test_cli_ignore_config_element` in `cli_test.rb`).

I'm used to python where an empty array as the default value of a parameter is a big gotcha thanks to mutable default arguments https://docs.python-guide.org/writing/gotchas/. I'll change that around.

Reply:
I think leaving this defaulting to `nil` is much of a muchness. It prevents the caller having to check for a value being provided or initializing the `attr_accessor` to an empty array. For example, if the default value is an array, the `ignored.each` call will raise an exception in the case that the called passes in `nil` to `Config.new(config, nil)`. This is likely because an option not provided to `OptionParser` will default to `nil` in the `Cli` class. Less checks are required just doing a nil guard here.